models/classes/search/ResultSetResponseNormalizer.php

```diff
@@ -70,18 +76,39 @@ public function normalize(SearchQuery $searchQuery, ResultSet $resultSet, string
                     $content['label'] = __('Access Denied');
                 }
 
+                $resource = $this->getResource($content['id']);
+
+                $readonly = false;
+
+                $topLevelClass = $this->getClass(TaoOntology::CLASS_URI_OBJECT);
+
+                foreach ($resource->getTypes() as $type) {
+                    $accessibleResources = $permissionHelper
+                    ->filterByPermission(
+                        array($type->getUri()),
+                        PermissionInterface::RIGHT_READ
+                    );
+
+                    if (empty($accessibleResources) || !$isAccessible) {
+                        $readonly = true;
+                        break;
+                    }
+
+                    $class = $this->getClass($type->getUri());
+                    $readonly = $this->checkParentClassPermission($class, $permissionHelper, $topLevelClass);
+
+                    if ($readonly === true) {
+                        break;
+                    }
+                }
+
                 $response['data'][] = $this->getResultSetFilter()->filter($content, $structure);
+
+                $readonlyArray[$content['id']] = $readonly;
```

Comment:
WDYT of changing name, might be something like that
```suggestion
                $readOnlyResources[$content['id']] = $readonly;
```